pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -301,6 +262,25 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder,
 	return resources, cacheStats{hits: hit, miss: miss}
 }
 
+func skipForDelta(key model.XdsCacheEntry, updated map[model.ConfigKey]struct{}) bool {
+	if len(updated) == 0 {
+		return false
+	}
+	for cfg := range updated {
+		for _, dt := range key.DependentTypes() {
+			if dt == cfg.Kind {
+				return false
+			}
+		}
+		for _, dc := range key.DependentConfigs() {
+			if dc == cfg {
+				return false
+			}
+		}
```

Comment:
Compared with, this is stateful, while that one can be seen as stateless. 

```
 func (cb *ClusterBuilder) getAllCachedSubsetClusters(clusterKey clusterCache) ([]*discovery.Resource, bool) {
	if !features.EnableCDSCaching {
		return nil, false
	}
	destinationRule := CastDestinationRule(clusterKey.destinationRule)
	res := make([]*discovery.Resource, 0, 1+len(destinationRule.GetSubsets()))
	cachedCluster, f := cb.cache.Get(&clusterKey)
	allFound := f
	res = append(res, cachedCluster)
	dir, _, host, port := model.ParseSubsetKey(clusterKey.clusterName)
	for _, ss := range destinationRule.GetSubsets() {
		clusterKey.clusterName = model.BuildSubsetKey(dir, ss.Name, host, port)
		cachedCluster, f := cb.cache.Get(&clusterKey)
		if !f {
			allFound = false
		}
		res = append(res, cachedCluster)
	}
	return res, allFound
}
```

Reply:
another point is that, the function looks not right. 
Look at `getAllCachedSubsetClusters`, the cluster key can be modified 
```
		clusterKey.clusterName = model.BuildSubsetKey(dir, ss.Name, host, port)

```

Reply:
Also, i am not sure if this function is more efficient than `getAllCachedSubsetClusters`